app/blueprints/threads/__init__.py

```diff
@@ -383,10 +383,15 @@ def new():
 
 
 @bp.route("/users/<username>/comments/")
-@rank_required(UserRank.EDITOR)
 def user_comments(username):
 	user = User.query.filter_by(username=username).first()
 	if user is None:
 		abort(404)
 
-	return render_template("threads/user_comments.html", user=user, replies=user.replies)
+	replies = [
+		reply
+		for reply in user.replies
+		if reply.check_perm(current_user, Permission.VIEW_REPLY)
```

Comment:
I think this should use SEE_THREAD on the Thread

Reply:
Do I understand correctly that the authorization should work like the following:
1. the user can see all his own comments on any thread
2. if the user is participating or staff he can see all comments on all threads
3. if the user is not the author of the comment and is not participating or staff he can see the comment if this comment is not on a private thread
4. previous rule is also valid for an unauthenticated user?
And the meaning of "user is participating or staff" is that the user is at least approver?

Reply:
Currently, there's no concept of per-comment visibility. A user can either see a thread or not. So it should be this:

```suggestion
		if reply.thread.check_perm(current_user, Permission.SEE_THREAD)
```

Reply:
I replaced the condition with SEE_THREAD permission check and also add selectinload to avoid N+1 problem.

Also, I successfully conducted the following test:
1. Create two users with usernames "one" and "two" and rank MEMBER
2. Create one public thread with one comment and one private thread with one comment, when logged in as user one
3. Check the following:
3.1. When logged in as user one, can see both comments
3.2. When logged in as user rubenwardy, can see both comments
3.3. When logged in as user two, can see only one comment on public thread